src/thread.rs

```diff
@@ -129,28 +129,28 @@ impl<T, F: FnOnce() -> T> FnBox<T> for F {
 }
 
 /// Like `std::thread::spawn`, but without lifetime bounds on the closure.
-pub unsafe fn spawn_unchecked<'a, F, T>(f: F) -> thread::JoinHandle<T>
+pub unsafe fn spawn_unchecked<'env, F, T>(f: F) -> thread::JoinHandle<T>
 where
     F: FnOnce() -> T,
-    F: Send + 'a,
+    F: Send + 'env,
     T: Send + 'static,
 {
     let builder = thread::Builder::new();
     builder_spawn_unchecked(builder, f).unwrap()
 }
 
 /// Like `std::thread::Builder::spawn`, but without lifetime bounds on the closure.
-pub unsafe fn builder_spawn_unchecked<'a, F, T>(
+pub unsafe fn builder_spawn_unchecked<'env, F, T>(
     builder: thread::Builder,
     f: F,
 ) -> io::Result<thread::JoinHandle<T>>
 where
     F: FnOnce() -> T,
-    F: Send + 'a,
+    F: Send + 'env,
     T: Send + 'static,
 {
-    let closure: Box<FnBox<T> + 'a> = Box::new(f);
-    let closure: Box<FnBox<T> + Send> = mem::transmute(closure);
+    let closure: Box<FnBox<T> + Send + 'env> = Box::new(f);
+    let closure: Box<FnBox<T> + Send + 'static> = mem::transmute(closure);
```

Comment:
These two lines seem receive an explicit lifetime bound as well, which should make it more clear what is going on here.
The "faking" of a `'static` lifetime bound for non-static references in order to use the closure in a `std::thread::Thread` should be more expressive this way.